### PR TITLE
Meta qt6 PR

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **qt/6.***

This PR gathers the content of the currently opened PRs, hoping to get some attention from the team:
- https://github.com/conan-io/conan-center-index/pull/22540
- https://github.com/conan-io/conan-center-index/pull/22700
- https://github.com/conan-io/conan-center-index/pull/21676
- https://github.com/conan-io/conan-center-index/pull/19631
- https://github.com/conan-io/conan-center-index/pull/22843

I only used merge commits, so merging this meta-PR automatically closes the PRs above. Also, if one of the PRs above is merged before this meta-PR, I'll simply merge master into this meta-PR and said PR content will disappear from this meta-PR

call for review @conan-io/barbarians

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7985589321/attempts/1)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7985589321/attempts/2)

File: recipes/qt/6.x.x/conanfile.py [hunk 1]
```diff
@@ -198,6 +200,7 @@ def configure(self):
             del self.options.with_libpng
             del self.options.with_md4c
             self.options.rm_safe("with_x11")
+            self.options.rm_safe("with_egl")
```
Comment (jwillikers) [inline recipes/qt/6.x.x/conanfile.py:203]: Should this option also be removed on macOS? I don't think EGL is available there.

**Review by jwillikers — APPROVED** [review]

File: recipes/qt/6.x.x/conanfile.py (hunk 1 above)
Comment (ericLemanissier) [reply recipes/qt/6.x.x/conanfile.py:203]: that's what line 169 does right ? (the change above the one you commented)

Comment (jwillikers) [reply recipes/qt/6.x.x/conanfile.py:203]: My bad, I missed that. Thanks!

**Review by MartinDelille — APPROVED** [review]

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7985589321/attempts/3)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8052987241/attempts/1)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8052987241/attempts/2)

**Review by jwillikers — APPROVED** [review]

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8052987241/attempts/3)

**Review by RubenRBS — APPROVED** [review]

**Comment by ericLemanissier** [comment]
thanks

**Pull request merged**